### PR TITLE
consume-config-ms

### PR DESCRIPTION
**Pull request by josesolaro**

### CODE REVIEW

File: Audit/Repository/AuditRepository.cs [hunk 1]
```diff
@@ -68,8 +71,18 @@ public async Task<bool> Delete(Entity.Audit doc)
 
         public async Task<bool> CreateDd()
         {
-            await ConnectionDB.CreateDatabaseAsync<Entity.Audit>().ConfigureAwait(true);
-            return true;
+            try
+            {
+                await ConnectionDB.CreateDatabaseAsync<Entity.Audit>().ConfigureAwait(true);
+                /*RestClient client = new RestClient(ConnectionDB.ConnectionString);
```
Comment (nestorreloaded) [inline Audit/Repository/AuditRepository.cs]: Please remove commented code (only should be use for documentation)

File: Audit/Repository/AuditRepository.cs [hunk 2]
```diff
@@ -81,7 +94,7 @@ public async Task<bool> IsUp()
                     return true;
                 }
                 return false;
-            }catch(CouchDB.Driver.Exceptions.CouchException e)
+            }catch(Exception e)
```
Comment (nestorreloaded) [inline Audit/Repository/AuditRepository.cs]: We should log exception messages

File: Audit/Services/ConfigService.cs [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        private Dictionary<string, string> ParseResponse(IRestResponse response)
+        {
+            if (response.StatusCode == System.Net.HttpStatusCode.OK)
+            {
+                var resp = SimpleJson.DeserializeObject<JsonObject>(response.Content);
+                return Utils.Utils.toDic(resp.ToString());
+            }
+            else
+            {
+                return null;
```
Comment (nestorreloaded) [inline Audit/Services/ConfigService.cs]: Please don't return null, it leads to a lot of Exceptions

File: Audit/Startup.cs [hunk 4]
```diff
@@ -99,10 +100,48 @@ static public void Configure(IApplicationBuilder app, IHostingEnvironment env, I
...
+            var my_conf = new Services.ConfigService().Get().Result;
+            while (my_conf == null)
+            {
+                //return Configuration;
+                System.Threading.Thread.Sleep(10000);
+                my_conf = new Services.ConfigService().Get().Result;
+            }
+            for (int i = 0; i < my_conf.Count; i++)
+            {
+                if (my_conf.ContainsKey("AUDIT:NOTTOSAVE:" + i))
```
Comment (nestorreloaded) [inline Audit/Startup.cs]: Please, put hardcoded strings into constants

File: Audit/Utils/Utils.cs [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+                                result.Add(jSub.Key.ToUpper() + ":" + jSub_resp.Key.ToUpper(), jSub_resp.Value.ToString());
+                            }
+                        }
+                    }
+                }
+                return result;
+            }
+            catch
+            {
+                return null;
```
Comment (nestorreloaded) [inline Audit/Utils/Utils.cs]: Please don't return null

File: Audit/Services/ConfigService.cs [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+        {
+            string endpoint = "/api/config/configuration/"+ Startup.StaticConfig["ENVIRONMENT"] +"-"+ Startup.StaticConfig["ENVIRONMENT"] + ".json";
+            var request = new RestRequest(endpoint, Method.GET);
+            request.AddHeader("Content-Type", "application/json");
+            return this.ParseResponse(await client.ExecuteTaskAsync(request).ConfigureAwait(true));
+        }
+
+        public async Task<object> Post(JsonParameter body)
+        {
+            return new object();
```
Comment (nestorreloaded) [inline Audit/Services/ConfigService.cs]: The body parameter isn't used in this method

File: Audit/Dockerfile [hunk 7]
```diff
@@ -1,13 +1,15 @@
 FROM mcr.microsoft.com/dotnet/core/aspnet:2.2-stretch-slim AS base
 ENV COUCHDB_URL=localhost \
+GATEWAY_URL=localhostossss:8080 \
```
Comment (nestorreloaded) [inline Audit/Dockerfile]: It shouln't be 'localhost' instead ?

**Review by nestorreloaded — CHANGES_REQUESTED** [review]
Please review the changes

**Review by nestorreloaded — APPROVED** [review]

**Review by exe135 — APPROVED** [review]

**Pull request merged**